src/main/java/com/scalar/db/benchmarks/tpcc/TpccBench.java

```diff
@@ -1,262 +1,159 @@
-      description = "The duration of benchmark in seconds")
-  private int duration;
-
-  @CommandLine.Option(
-      names = {"--ramp-up-time"},
-      paramLabel = "RAMP_UP_TIME",
-      defaultValue = "60",
-      description = "The ramp up time in seconds.")
-  private int rampUpTime;
-
-  @CommandLine.Option(
-      names = {"--times"},
-      paramLabel = "TIMES",
-      defaultValue = "0",
-      description = "The number of serial execution for testing.")
-  private int times;
-
-  @CommandLine.Option(
-      names = {"--backoff"},
-      paramLabel = "BACKOFF",
-      defaultValue = "0",
-      description = "The milliseconds of backoff when retrying.")
-  private int backoff;
-
-  @CommandLine.Option(
-      names = {"--use-table-index"},
-      paramLabel = "USE_TABLE_INDEX",
-      defaultValue = "false",
-      description = "Use table-based secondary index.")
-  private boolean useTableIndex;
-
-  @CommandLine.Option(
-      names = {"--verbose"},
-      paramLabel = "VERBOSE",
-      defaultValue = "false",
-      description = "Show verbose error messages.")
-  boolean verbose;
-
-  @CommandLine.Option(
-      names = {"-h", "--help"},
-      usageHelp = true,
-      description = "display the help message.")
-  private boolean helpRequested;
-
-  private final AtomicInteger counter = new AtomicInteger();
-  private final AtomicInteger totalCounter = new AtomicInteger();
-  private final AtomicLong latencyTotal = new AtomicLong();
-  private final AtomicInteger errorCounter = new AtomicInteger();
-
-  public static void main(String[] args) {
-    int exitCode = new CommandLine(new TpccBench()).execute(args);
-    System.exit(exitCode);
-  }
-
-  @Override
-  public Integer call() throws Exception {
-    DatabaseConfig dbConfig = new DatabaseConfig(new FileInputStream(properties));
+import javax.json.Json;
+
+public class TpccBench extends TimeBasedProcessor {
+  private static final String CONFIG_NAME = "tpcc_config";
+  private static final String NUM_WAREHOUSES = "num_warehouses";
+  private static final String BACKOFF = "backoff";
+  private static final String USE_TABLE_INDEX = "use_table_index";
+  private static final String NP_ONLY = "np_only";
+  private static final String RATE_NEW_ORDER = "rate_new_order";
+  private static final String RATE_PAYMENT = "rate_payment";
+  private static final String RATE_ORDER_STATUS = "rate_order_status";
+  private static final String RATE_DELIVERY = "rate_delivery";
+  private static final String RATE_STOCK_LEVEL = "rate_stock_level";
+  private static final long DEFAULT_NUM_WAREHOUSES = 1;
+  private static final long DEFAULT_BACKOFF = 0;
+  private static final boolean DEFAULT_USE_TABLE_INDEX = false;
+  private final DistributedTransactionManager manager;
+  private final AtomicInteger abortCounter = new AtomicInteger();
+  private final TpccConfig tpccConfig;
+  private final int numWarehouses;
+  private final int backoff;
+  private final boolean useTableIndex;
```

Comment:
It looks like these fields can be local variables.

Reply:
Fixed in 2286bc9.